src/Marten/Events/Projections/Flattened/FlatTableProjection.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Baseline;
+using LamarCodeGeneration;
+using Marten.Events.Daemon;
+using Marten.Linq.Parsing;
+using Marten.Storage;
+using Weasel.Core;
+using Weasel.Postgresql.Tables;
+
+#nullable enable
+
+namespace Marten.Events.Projections.Flattened
+{
+    /// <summary>
+    /// Projection type that will write event data to a single database table
+    /// </summary>
+    public partial class FlatTableProjection : GeneratedProjection, IProjectionSchemaSource
+    {
+        private readonly List<IEventHandler> _handlers = new ();
+        private readonly string _inlineTypeName;
+
+        public FlatTableProjection(string tableName, SchemaNameSource schemaNameSource): this(new DbObjectName("public", tableName), schemaNameSource)
+        {
+
+        }
+
+        public FlatTableProjection(DbObjectName tableName) : this(tableName, SchemaNameSource.Explicit){}
+
+        private FlatTableProjection(DbObjectName tableName, SchemaNameSource schemaNameSource): base(tableName.Name)
+        {
+            SchemaNameSource = schemaNameSource;
+            Table = new Table(tableName);
+            _inlineTypeName = GetType().ToSuffixedTypeName("InlineProjection");
+
+            _generatedProjection = new Lazy<IProjection>(() =>
+            {
+                if (_generatedType == null)
+                {
+                    throw new InvalidOperationException("The EventProjection has not created its inner IProjection");
+                }
+
+                var projection = (IProjection)Activator.CreateInstance(_generatedType)!;
+
+                return projection!;
+            });
+        }
+
+        public SchemaNameSource SchemaNameSource { get; }
+
+        /// <summary>
+        /// The definition of the table being written to. You can use this to
+        /// modify the table structure or even add indexes
+        /// </summary>
+        public Table Table { get; }
+
+        /// <summary>
+        /// Write values for the designated event type to the table columns
+        /// </summary>
+        /// <param name="configure"></param>
+        /// <param name="tablePrimaryKeySource">If specified, tells the projection how to find the primary key value for the table from the event data. If missing, the projection uses the event's stream identity by default</param>
+        /// <typeparam name="T"></typeparam>
+        public void Project<T>(Action<StatementMap<T>> configure, Expression<Func<T, object>>? tablePrimaryKeySource = null)
```

Comment:
**QUESTION:** We'll support the right now single result value, right? (Not e.g. multiple identities)?

If we add custom SQL then it might be fine for people to not provide identity, but then that's probably a different projection type.

Reply:
Like I said before, if you get out of the well lit path, you can go to an `EventProjection` and write custom SQL to your heart's content. Same with the aggregations, I don't want to even attempt to accommodate everything they could try to do in the OOTB recipe because that kills us with permutations. Let them go to custom, explicit code for anything weird or unusual.